Srapolu/dropbear

s-rapolu opened the pull request:
Dropbear plugin implementation related changes in buildroot repository.

s-rapolu commented:
1. Made changes to the native implementation to support changes required for configuration of the service
2. Updated required files to consider changes done for SecureShellServer interface.

wouterlucas commented on package/dropbear/0001-Makefile.in-changes.patch.conditional:
```diff
@@ -0,0 +1,76 @@
+--- a/Makefile.in	2016-07-21 20:47:09.000000000 +0530
```
Patch needs appropriate format-patch header, including commit log providing a reason for the patch. For more info see: https://git-scm.com/docs/git-format-patch

s-rapolu replied:
Yes, made the corrections and now created the patch with git format-patch header.

wouterlucas commented on package/dropbear/0002-svr-main.c-changes.patch.conditional:
```diff
@@ -0,0 +1,398 @@
+--- a/svr-main.c	2016-07-21 20:47:09.000000000 +0530
```
Patch needs appropriate format-patch header, including commit log providing a reason for the patch. For more info see: https://git-scm.com/docs/git-format-patch

s-rapolu replied:
Yes, made the corrections and now created the patch with git format-patch header.

wouterlucas commented on package/dropbear/0003-configure.ac-changes.patch.conditional:
```diff
@@ -0,0 +1,12 @@
+--- a/configure.ac	2016-07-21 20:47:09.000000000 +0530
```
Patch needs appropriate format-patch header, including commit log providing a reason for the patch. For more info see: https://git-scm.com/docs/git-format-patch

s-rapolu replied:
Yes, made the corrections and now created the patch with git format-patch header.

wouterlucas commented on package/dropbear/0004-libdropbear.h-added.patch.conditional:
```diff
@@ -0,0 +1,31 @@
+--- a/libdropbear.h	2019-08-28 20:32:09.684490106 +0530
```
Patch needs appropriate format-patch header, including commit log providing a reason for the patch. For more info see: https://git-scm.com/docs/git-format-patch

s-rapolu replied:
Yes, made the corrections and now created the patch with git format-patch header.

wouterlucas commented on package/dropbear/0005-libdropbear.pc.in-added.patch.conditional:
```diff
@@ -0,0 +1,14 @@
+--- a/libdropbear.pc.in	1970-01-01 05:30:00.000000000 +0530
```
Patch needs appropriate format-patch header, including commit log providing a reason for the patch. For more info see: https://git-scm.com/docs/git-format-patch

s-rapolu replied:
Yes, made the corrections and now created the patch with git format-patch header.

wouterlucas commented on package/dropbear/0006-linkedlist.h-added.patch.conditional:
```diff
@@ -0,0 +1,172 @@
+--- a/linkedlist.h	2019-07-11 14:07:04.040000000 +0530
```
Patch needs appropriate format-patch header, including commit log providing a reason for the patch. For more info see: https://git-scm.com/docs/git-format-patch

s-rapolu replied:
Yes, made the corrections and now created the patch with git format-patch header.

wouterlucas commented on package/wpe/wpeframework-plugins/Config.in:
```diff
@@ -603,6 +603,13 @@ config BR2_PACKAGE_WPEFRAMEWORK_SYSTEMCOMMANDS
       SystemCommands Plugin
 
+config BR2_PACKAGE_WPEFRAMEWORK_SECURESHELLSERVER
+    select BR2_PACKAGE_WPEFRAMEWORK_PLUGINS
```
I dont think this is needed, you're already in the wpeframework-plugins config right?

s-rapolu replied:
Yes, removed.

wouterlucas commented on package/wpe/wpeframework-plugins/wpeframework-plugins.mk:
```diff
@@ -4,8 +4,8 @@
-WPEFRAMEWORK_PLUGINS_VERSION = 8b70bd116e65f8824f060e0e26a4b1cfe65e27cd
-WPEFRAMEWORK_PLUGINS_SITE = $(call github,WebPlatformForEmbedded,WPEFrameworkPlugins,$(WPEFRAMEWORK_PLUGINS_VERSION))
+WPEFRAMEWORK_PLUGINS_VERSION = 5d0769cea06138762ac8ecc102b5b8e5c9b3a863
+WPEFRAMEWORK_PLUGINS_SITE = $(call github,s-rapolu,WPEFrameworkPlugins,$(WPEFRAMEWORK_PLUGINS_VERSION))
```
We cant point to your fork, this needs to be a commit on the main repository (master) that has the required changes to support the ssh server plugin.

s-rapolu replied:
Sorry, corrected the mistake and now pointing it to master.

wouterlucas commented on package/wpe/wpeframework-plugins/wpeframework-plugins.mk:
```diff
@@ -72,6 +72,17 @@ ifeq ($(BR2_PACKAGE_WPEFRAMEWORK_DICTIONARY),y)
 WPEFRAMEWORK_PLUGINS_CONF_OPTS += -DPLUGIN_DICTIONARY=ON
 endif
 
+ifeq ($(BR2_PACKAGE_WPEFRAMEWORK_SECURESHELLSERVER),y)
```
Can you move this down to other S related selections? Trying to organize this A-Z for readability.

s-rapolu replied:
Yes. Corrected.

wouterlucas commented on package/wpe/wpeframework/wpeframework.mk:
```diff
@@ -4,8 +4,8 @@
-WPEFRAMEWORK_VERSION = aed13867c93fe76dc29d32abe4237559cc19b62c
-WPEFRAMEWORK_SITE = $(call github,WebPlatformForEmbedded,WPEFramework,$(WPEFRAMEWORK_VERSION))
+WPEFRAMEWORK_VERSION = 2c1a62d177473c67c1abbd41147a051edb436433
+WPEFRAMEWORK_SITE = $(call github,s-rapolu,WPEFramework,$(WPEFRAMEWORK_VERSION))
```
Same here, cant point the entire buildroot to your fork. Needs to point to master with your changes merged in on the main repo.

s-rapolu replied:
Sorry, corrected the mistake and now pointing it to master.

s-rapolu commented:
1. Fixed issue with git format-patch
2. Removed the reference of forked repository and instead pointing it to now main repositories.
3. Fixed issues with signal handling in native dropbear session management
4. Organized the SSH plugin placements in corresponding files for readability.

Please review it and let us know your feedback.
